sentry_streams/sentry_streams/runner.py

```diff
@@ -39,9 +39,14 @@ def iterate_edges(p_graph: Pipeline, translator: RuntimeTranslator[Stream, Strea
                 for output in output_steps:
                     next_step: WithInput = cast(WithInput, p_graph.steps[output])
                     print(f"Apply step: {next_step.name}")
-                    # TODO: Make the typing align with the streams being iterated through. Reconsider algorithm as needed.
                     next_step_stream = translator.translate_step(next_step, input_stream)  # type: ignore
-                    step_streams[next_step.name] = next_step_stream
+                    # if next_step_stream is a map of downstream routes
+                    if isinstance(next_step_stream, MutableMapping):
+                        for branch_name in next_step_stream:
+                            step_streams[branch_name] = next_step_stream[branch_name]
+                    else:
+                        # TODO: Make the typing align with the streams being iterated through. Reconsider algorithm as needed.
```

Comment:
nit, This TODO comment had to do with the # type: ignore I left above, can you move it there please